configuration_service/broadcast_message.go

```diff
@@ -32,7 +32,7 @@ func (broadcast *MessageBroadcaster) NewSubscriber() chan int {
 		broadcast.subscribers = make([]chan int,0)
 	}
 	broadcast.subscribers = append(broadcast.subscribers, ch)
-	broadcast.mutex.Unlock()
+	defer broadcast.mutex.Unlock()
```

Comment:
If error happens between `Lock()` and `defer Unlock()` lines, then defer operator is not be executed and lock is not released. So `defer broadcast.mutex.Unlock()` line should be located just after `broadcast.mutex.Lock()` (same for `Publish` method)

Reply:
My Bad :(  fixing this now right away